.github/workflows/push.yaml

```diff
@@ -10,70 +10,75 @@ on:
 
 jobs:
   docker-build:
-    runs-on: ubuntu-20.04
+    runs-on: ec2-runners
+    container:
+      image: ${{ vars.EC2_IMAGE }}
     timeout-minutes: 20
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       - name: Configure AWS credentials
-        uses: aws-actions/configure-aws-credentials@v1
+        # See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#requesting-the-access-token
+        uses: aws-actions/configure-aws-credentials@v4
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: eu-west-1
 
       - name: Login to Amazon ECR
         id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+        uses: aws-actions/amazon-ecr-login@v2
 
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
 
       - name: Cache Docker layers
-        uses: actions/cache@v2
+        uses: actions/cache@v4
         with:
           path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-buildx-cache-v2-${{ github.sha }}
+          key: ${{ runner.os }}-buildx-cache-${{ github.sha }}
           restore-keys: |
-            ${{ runner.os }}-buildx-cache-v2
+            ${{ runner.os }}-buildx-cache
 
       - name: Build and push
         uses: docker/build-push-action@v2
         with:
           context: .
           push: true
-          tags: "${{ env.DOCKER_REGISTRY }}/dss:${{ github.sha }},${{ env.DOCKER_REGISTRY }}/dss:latest"
+          tags: "${{ vars.CONTAINER_REGISTRY }}/dss:${{ github.sha }},${{ vars.CONTAINER_REGISTRY }}/dss:latest"
           cache-from: type=local,src=/tmp/.buildx-cache
           cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
-      - uses: nelonoel/branch-name@v1.0.1
+
       - name: Trigger Deployment for service
         # if: github.ref == 'refs/heads/main'
         env:
           SPINNAKER_WEBHOOK_TOKEN: ${{ secrets.SPINNAKER_WEBHOOK_TOKEN }}
         run: |
-          echo "Sending webhook with branch: ${BRANCH_NAME}"
+          set -x
+          echo "Sending webhook with branch: ${GITHUB_REF_NAME}"
           curl $SPINNAKER_URL/webhooks/webhook/digital-signature-service -H "content-type: application/json" --data-binary @- <<BODY
           {
             "token": "$SPINNAKER_WEBHOOK_TOKEN",
-            "branch": "${BRANCH_NAME}",
+            "branch": "${GITHUB_REF_NAME}",
             "parameters": {
               "docker_tag": "${{ github.sha }}",
-              "feature_name": "$(echo "${BRANCH_NAME}" | cut -d"/" -f2- | tr -cd '[:alnum:].-' | tr '[:upper:]' '[:lower:]' | cut -c1-50)",
+              "feature_name": "$(echo "${GITHUB_REF_NAME}" | cut -d"/" -f2- | tr -cd '[:alnum:].-' | tr '[:upper:]' '[:lower:]' | cut -c1-50)",
```

Comment:
`GITHUB_REF_NAME` doesn't return a meaningful branch name in PRs, [see docs](https://docs.github.com/en/actions/learn-github-actions/variables).

The job demonstrates this
https://github.com/island-is/digital-signature-service/actions/runs/8687258075/job/23820415231#step:9:33

Reply:
Ah, my bad
> For pull requests, the format is <pr_number>/merge.